src/core/algorithms/fd/hyfd/model/fd_tree_vertex.cpp

```diff
@@ -9,16 +9,16 @@ namespace algos::hyfd::fd_tree {
 
 void FDTreeVertex::GetLevelRecursive(unsigned target_level, unsigned cur_level,
                                      boost::dynamic_bitset<> lhs, std::vector<LhsPair>& vertices) {
-    if (cur_level == target_level) {
+    if (cur_level == target_level && fds_.any()) {
```

Comment:
```suggestion
    if (cur_level == target_level) {
```
Move to line below: `if (fds_.any()) vertices. ...`